### ISSUE TITLE
CellProfiler 4.0.7 freezes with last image sets

### ISSUE THREAD

**Issue by lakshmibalasubramanian** [CellProfiler/CellProfiler#4358]
Hi,

I tried to extract the maximum possible features from the pipeline that I made using CellPainting assay dataset. The pipeline just freezes at the end of the image list and doesn't complete. There is no error message as well. This is the same issue that is already posted in the forum [here](https://forum.image.sc/t/cellprofiler-4-freezes-near-end-of-list-of-image-sets/43098/15). 
With the small number of images & reduced the measurements it worked fine.

@bethac07 
Here is the drive link for the dataset & the pipeline,
https://drive.google.com/drive/folders/1D2pMM8BSLJKuGDrvjDrS-m_1XdUyY1s7?usp=sharing

Hope you are able to reproduce the error.

Regards,
Lakshmi

**Comment by bethac07** [comment]
The sample image set you sent contained almost 3500 images, but your cpproj only shows about 50 loaded- which is the minimal size to trigger this issue? Have you tested whether using ExportToDatabase vs ExportToSpreadsheet adjust the behavior?

If the issue is a whole plate of profiling not completing on one machine, I'm not surprised, particularly with ExportToSpreadsheet; for a plate of data we routinely see total measurement sizes of ~40 GB of measurements, which in the case of using ExportToSpreadsheet, are first all dumped into a temporary file and then have to be written to the spreadsheet.  We would certainly recommend running on a cluster at that scale.

Please do clarify what we need to do to test this on our end.  CellProfiler version, OS version, and the build method (source vs prebuilt) might prove useful as well.

**Comment by lakshmibalasubramanian** [comment]
@bethac07 
Yes, I was trying with few set of images i.e.50.  I wanted to try 100 images at least, of course I don't want to try for a complete plate in one machine.
I have not tested ExportToDatabase vs ExportToSpreadsheet though I used only spreadsheet.
I tried in Cellprofiler4.0.7 in windows where I had a problem.

**Comment by bethac07** [comment]
I've tried this in a few situations on my mac and could not reproduce with 48 image sets:

4.0.7 built
4.1.3 built
4.1.3 source
4.1.3 source, using an "Elsewhere" location rather than Default Output (your pipeline had an "Elsewhere" location, but for other runs I had to switch it because your location was of course not a location on my computer)
I am going to try running overnight with 240 data sets in 4.0.7 built/ 4.1.3 built/4.1.3 source to see if I can reproduce with a larger number of sets; if I can't, it's possible it's maybe a Windows specific thing, or to do with some other aspect of the run environment (number of workers perhaps, I'm using only 1 so I could test multiple versions alongside one another).

**Comment by bethac07** [comment]
Cannot validate with 240 image sets either.

**Comment by lakshmibalasubramanian** [comment]
ok. Thanks for the trial. May be it could be the windows issue.
